spec/ioki/apis/endpoints/endpoints_spec.rb

```diff
@@ -3,33 +3,36 @@
 RSpec.describe Endpoints do
   describe '.url_elements' do
     it 'creates a concrete path from one with placeholders and corresponding string args' do
-      expect(described_class.url_elements('platform', [:id], 'RIDE_ID')).
-        to eq ['RIDE_ID']
+      expect(described_class.url_elements([:id], 'RIDE_ID')).to eq ['RIDE_ID']
     end
 
     it 'keeps path fragments that are not meant to be interpolated' do
-      expect(described_class.url_elements('platform', ['ride', :id], 'RIDE_ID')).
-        to eq %w[ride RIDE_ID]
+      expect(described_class.url_elements(['ride', :id], 'RIDE_ID')).to eq %w[ride RIDE_ID]
     end
 
     it 'calls the interpolation method on the given arguments' do
       expect(
-        described_class.url_elements(
-          'platform',
-          ['ride', :id],
-          instance_double(Ioki::Model::Platform::Ride, 'ride', id: 'RIDE_ID')
-        )
+        described_class.url_elements(['ride', :id], instance_double(Ioki::Model::Platform::Ride, 'ride', id: 'RIDE_ID'))
       ).to eq %w[ride RIDE_ID]
     end
 
     it 'raises an error if the argument does not respond_to the interpolation method' do
-      expect do
-        described_class.url_elements(
-          'platform',
-          [:id],
-          instance_double(Ioki::Model::Platform::RidePassenger, 'non_id_object')
-        )
-      end.to raise_error(ArgumentError, "#'platform': argument 1 must respond to :id or to be a string")
+      expect { described_class.url_elements([:id], 1) }.
+        to raise_error(NoMethodError, /undefined method `id' for 1:Integer/)
```

Comment:
I think the NoMethodError is as good as the custom error here and it allows to simplify the code a bit.